pkgs/development/libraries/tk/generic.nix

```diff
@@ -19,8 +19,13 @@ stdenv.mkDerivation {
   postInstall = ''
     ln -s $out/bin/wish* $out/bin/wish
     cp ../{unix,generic}/*.h $out/include
+  ''
+  + stdenv.lib.optionalString (stdenv.isLinux) ''
     ln -s $out/lib/libtk${tcl.release}.so $out/lib/libtk.so
   ''
+  + stdenv.lib.optionalString (stdenv.isDarwin) ''
+    ln -s $out/lib/libtk${tcl.release}.dylib $out/lib/libtk.dylib
+  ''
```

Comment:
please use `stdenv.hostPlatform.extensions.sharedLibrary`

```diff
--- i/pkgs/development/libraries/tk/generic.nix
+++ w/pkgs/development/libraries/tk/generic.nix
@@ -19,7 +19,7 @@ stdenv.mkDerivation {
   postInstall = ''
     ln -s $out/bin/wish* $out/bin/wish
     cp ../{unix,generic}/*.h $out/include
-    ln -s $out/lib/libtk${tcl.release}.so $out/lib/libtk.so
+    ln -s $out/lib/libtk${tcl.release}${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/libtk${stdenv.hostPlatform.extensions.sharedLibrary}
   ''
   + stdenv.lib.optionalString (stdenv.isDarwin) ''
     cp ../macosx/*.h $out/include
```

Reply:
Thanks for pointing that out. I wasn't aware of extensions.sharedLibrary, but I felt kinda dirty implementing it the way I did ;)